graylog2-web-interface/src/components/throughput/NodeThroughput.jsx

```diff
@@ -1,60 +1,70 @@
-                }
-                // update is [{nodeId, values: [{name, value: {metric}}]} ...]
-                // metric can be various different things, depending on metric {type: "GAUGE"|"COUNTER"|"METER"|"TIMER"}
-                var throughIn = 0;
-                var throughOut = 0;
-                // we will only get 0 or 1 node here.
-                update.forEach((perNode) => {
-                    perNode.values.forEach((namedMetric) => {
-                        if (namedMetric.name === "org.graylog2.throughput.input.1-sec-rate") {
-                            throughIn += namedMetric.metric.value;
-                        } else if (namedMetric.name === "org.graylog2.throughput.output.1-sec-rate") {
-                            throughOut += namedMetric.metric.value;
-                        }
-                    });
-                });
-                this.setState({initialized: true, totalIn: throughIn, totalOut: throughOut, hasError: hasError});
-            }
-        });
-    },
-    render() {
-        if (this.state.hasError) {
-            return (<span>Unable to load throughput.</span>);
-        }
-        if (!this.state.initialized) {
-            return (<span><i className="fa fa-spin fa-spinner"></i> Loading throughput...</span>);
-        }
-        return (<span>
-            Processing <strong>{numeral(this.state.totalIn).format("0,0")}</strong> incoming and <strong>
-            {numeral(this.state.totalOut).format("0,0")}</strong> outgoing msg/s.
-        </span>);
+const NodeThroughput = React.createClass({
+  propTypes: {
+    nodeId: PropTypes.string.isRequired,
+    longFormat: PropTypes.bool,
+  },
+  mixins: [Reflux.connect(MetricsStore)],
+  getDefaultProps() {
+    return {
+      longFormat: false,
+    };
+  },
+  componentWillMount() {
+    this.metricNames = {
+      totalIn: 'org.graylog2.throughput.input.1-sec-rate',
+      totalOut: 'org.graylog2.throughput.output.1-sec-rate',
+    };
+
+    Object.keys(this.metricNames).forEach(metricShortName => MetricsActions.add(this.props.nodeId, this.metricNames[metricShortName]));
+  },
+  componentWillUnmount() {
+    Object.keys(this.metricNames).forEach(metricShortName => MetricsActions.remove(this.props.nodeId, this.metricNames[metricShortName]));
+  },
+  _isLoading() {
+    return !this.state.metrics;
+  },
+  _formatThroughput(metrics) {
+    if (this.props.longFormat) {
+      return (
+        <span>
+          Processing <strong>{numeral(metrics.totalIn).format('0,0')}</strong> incoming and <strong>
+          {numeral(metrics.totalOut).format('0,0')}</strong> outgoing msg/s.
+        </span>
+      );
+    } else {
+      return (
+        <span>
+          In {numeral(metrics.totalIn).format('0,0')} / Out {numeral(metrics.totalOut).format('0,0')} msg/s.
+        </span>
+      );
     }
+  },
+  render() {
+    if (this._isLoading()) {
+      return (<span><i className="fa fa-spin fa-spinner"/> Loading throughput...</span>);
```

Comment:
You can reuse the components/common/Spinner component for this and pass a text prop.